six/modules/c++/cphd/include/cphd/CPHDXMLControl.h

```diff
@@ -46,75 +54,135 @@ class CPHDXMLControl : public six::XMLParser
     //!  Constructor
     CPHDXMLControl();
 
-    CPHDXMLControl(logging::Logger* log, bool ownLog = false);
+    //! Constructor
+    CPHDXMLControl(logging::Logger* log, bool ownLog);
 
-    /*!
-     *  This function takes in a Metadata object and converts
-     *  it to a new-allocated XML DOM.
-     */
-    std::auto_ptr<xml::lite::Document> toXML(const Metadata& metadata);
```

Comment:
The function toXMLString calls toXML and prints out the document so I didn't think toXML itself would need to be public.
Wouldn't hurt to make it public though, in case anyone wants to access the document object itself.